content/context-resource/index.md

```diff
@@ -0,0 +1,320 @@
+---
+rfc: 0018
+start_date: 2018-08-13
+pr: openregister/registers-rfcs#31
+status: draft
+---
+
+# Context resource
+
+## Summary
+
+This RFC proposes a context resource to surface the metadata of the register.
+
+Meanwhile there is no versioning defined for metadata treat this as single
+versioned.
+
+Dependencies:
+
+* [RFC0013: Multihash](https://github.com/openregister/registers-rfcs/pull/26)
+
+
+## Motivation
+
+Currently, the register metadata is not exposed consistently. Some is exposed
+via the register resource, some via the field register and some from the
+register register.
+
+Also, the current specification and implementation don't agree on what is the
+register resource.
+
+The main driver for this RFC is to have a computed resource for the latest
+metadata in a way that makes the register self contained (i.e. a user should
+not need to interact with any other register to get the metadata for a
+register).
+
+
+## Explanation
+
+The **context** is the metadata snapshot that apply to a given log size.
+
+### Definition
+
+```elm
+type Context =
+  { id : Name
+  , copyright : String
+  , custodian : Maybe String
+  , description : Maybe String
+  , hashingAlgorithm : HashingAlgorithm
+  , licence : Maybe String
+  , rootHash : Hash
+  , schema : Schema
+  , stats : Stats
+  , status : Status
+  , title : Maybe String
+  }
+```
+
+See each section below of the explanation for each attribute.
+
+
+#### Id
+
+* Type: Name (This is defined in the specification as `[a-z][a-z0-9-]*`, the
+  future specification will make it more clear).
+
+The register identifier. The current implications for this value make it act
```

Comment:
What do you mean by "current implications"?

I'm not sure if this makes sense, but I think the environment it belongs to is still defined by the register or register (or another catalog of registers), which makes me wonder whether all of this metadata actually needs to be exposed through the API.

Things like title, description and licence seem very catalogue-y in the sense that they are relevant when deciding whether to use a register, but if you're already consuming from one then it doesn't seem important.

Reply:
The goal is to make a register self descriptive. Personally, I would like to not have a unique ID for a register within a register but it is how things are right now.

On title, description, again the goal is to have a self contained register such that it can be transported with minimal lose.

The catalogue (register register) right now duplicates this information and it is something we will need to address in the future.